local_planner/src/local_planner.cpp

```diff
@@ -281,27 +281,18 @@ void LocalPlanner::getStateAndReferencePlan() {
   ref_ground_height_(0) = local_footstep_planner_->getTerrainHeight(
       current_state_(0), current_state_(1));
 
-  // Update the body plan to use for linearization
-  if (body_plan_.rows() < N_) {
+  // Update the body plan to use for foot planning
+  int N_current_plan = body_plan_.rows();
+  if (N_current_plan < N_) {
     // Cold start with reference  plan
-    body_plan_ = ref_body_plan_;
+    body_plan_.conservativeResize(N_, 12);
 
     // Initialize with the current foot positions
-    for (int i = 0; i < N_; i++) {
+    for (int i = N_current_plan; i < N_; i++) {
+      body_plan_.row(i) = ref_body_plan_.row(i);
       foot_positions_body_.row(i) = current_foot_positions_body_;
       foot_positions_world_.row(i) = current_foot_positions_world_;
     }
-  } else {
-    // Only shift the foot position if it's a solve for a new plan index
-    if (!same_plan_index_) {
-      body_plan_.topRows(N_ - 1) = body_plan_.bottomRows(N_ - 1);
-      grf_plan_.topRows(N_ - 2) = grf_plan_.bottomRows(N_ - 2);
-
-      foot_positions_body_.topRows(N_ - 1) =
-          foot_positions_body_.bottomRows(N_ - 1);
-      foot_positions_world_.topRows(N_ - 1) =
-          foot_positions_world_.bottomRows(N_ - 1);
-    }
```

Comment:
Why do you delete the shift? I think we need it for the footstep plan.

Reply:
Hmm, looks like some bug got introduced after I got this working. Will update once fixed.

Reply:
Okay, updated. The bug was that I allowed NMPC to overwrite the foot plan data but allowed it to change the length of the foot plan rather than just overwriting the old data. Returned this section to its former state (you are right that this is used for the new foot plan construction).